src/ZoneServer/Skills/Handlers/Swordsman/Restrain.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Melia.Shared.L10N;
+using Melia.Shared.Game.Const;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+
+namespace Melia.Zone.Skills.Handlers.Swordsman
+{
+	/// <summary>
+	/// Handler for the Swordman skill Restrain
+	/// </summary>
+	[SkillHandler(SkillId.Swordman_Restrain)]
+	public class Restrain : ISelfSkillHandler
+	{
+		private const float ChancePerLevel = 6f; // 6% per level
+
+		/// <summary>
+		/// Handles skill, applying the buff to the caster.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="dir"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Direction dir)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.SetAttackState(true);
+
+			var target = caster;
+
+			var duration = TimeSpan.FromSeconds(30 + 3 * skill.Level);
+			// num2 is stun chance on normal attack
```

Comment:
See previous comment about comments on buff parameters. Curiously you used two different names to describe these parameters, and neither matches the actual name, `NumArg2`, adding to the potential confusion.

Reply:
It all makes perfect sense in my head.  I've changed the format to match Concentrate.